cli/module_registry.go

```diff
@@ -316,6 +332,30 @@ func UploadModuleAction(c *cli.Context) error {
 	return nil
 }
 
+// call validateModelAPI on all models in manifest and warn if violations.
+func validateModels(errWriter io.Writer, manifest *moduleManifest) {
+	for _, model := range manifest.Models {
+		if err := validateModelAPI(model.API); err != nil {
+			warningf(errWriter, "error validating API string %s: %s", model.API, err)
+		}
+	}
+}
+
+// return a useful error if the model string looks wrong.
+func validateModelAPI(modelAPI string) error {
+	api, err := resource.ParseAPIString(modelAPI)
+	if err != nil {
+		return errors.Wrap(err, "unparseable model string")
+	}
+	if err := api.Validate(); err != nil {
+		return errors.Wrap(err, "failed to validate API")
+	}
+	if !slices.Contains(rdkAPITypes, api.Type.Name) {
```

Comment:
[opt] use `api.IsComponent() || api.IsService()` like the other uses in rdk

Reply:
if the `rdkAPITypes` slice lived in RDK instead of CLI, it would be a canonical place to store the check; it's self-updating in a way that the `||` expression isn't

Reply:
Or better yet, if it lived inside `api.Validate()`

Reply:
or like generally would be nice if go enums exposed the entire list of values at the language level

we sort of get this from the linter but only in switch statements

Reply:
> if it lived inside api.Validate()

yes good point